hnn_core/tests/test_gui.py

```diff
@@ -563,3 +565,68 @@ def add_child_decorated(self, to_add):
     # Check if the widgets are relinked, the selected index should be synced
     gui.tab_group_1.selected_index = 0
     assert gui.tab_group_2.selected_index == 0
+
+
+def test_gui_download_simulation():
+    """Test the GUI download simulation pipeline."""
+    gui = HNNGUI()
+    _ = gui.compose()
+    gui.params['N_pyr_x'] = 3
+    gui.params['N_pyr_y'] = 3
+
+    # Run a simulation with 3 trials
+    gui.widget_dt.value = 0.85
+    gui.widget_ntrials.value = 2
+
+    # Initiate 1rs simulation
+    sim_name = "sim1"
+    gui.widget_simulation_name.value = sim_name
+
+    # Run simulation
+    gui.run_button.click()
+
+    _, file_extension = (
+        serialize_simulation(gui.data, sim_name))
+    # result is a zip file
+    assert file_extension == ".zip"
+
+    # Run a simulation with 1 trials
+    gui.widget_dt.value = 0.85
+    gui.widget_ntrials.value = 1
+
+    # Initiate 2nd simulation
+    sim_name2 = "sim2"
+    gui.widget_simulation_name.value = sim_name2
+
+    # Run simulation
+    gui.run_button.click()
+    _, file_extension = (
+        serialize_simulation(gui.data, sim_name2))
+    # result is a single csv file
+    assert file_extension == ".csv"
+
+
+def test_gui_upload_csv_simulation():
+    """Test if gui handles uploaded csv data"""
+    gui = HNNGUI()
+    _ = gui.compose()
+
+    assert len(gui.viz_manager.data['figs']) == 0
+    assert len(gui.data['simulation_data']) == 0
+
+    # Relative path to the file
+    file_relative_path = "./hnn_core/tests/assets/test_default.csv"
```

Comment:
possible to add an error raising test when the extension is not `.csv` or `.txt`?

Reply:
The logic for uploading csv files is implemented entirely in the GUI layer, and the FileUpload widget already filters files by type. This test programmatically bypasses that filter, but I cant see a scenario where someone would attempt to upload files other than txt or csv.